backend/src/mocks/chainArchive.ts

```diff
@@ -1,15 +1,17 @@
-import ChainArchive, { ArchivedBlock } from '../chainArchive';
+// import { Logger } from "pino";
 
+import ChainArchive, { ArchivedBlock } from '../chainArchive';
 import * as signedBlockMock from '../mocks/signedBlock.json';
 import { blockToBinary } from '../utils';
+// import loggerMock from './logger';
 
 export const blocksMock = [
   blockToBinary(signedBlockMock),
   blockToBinary(signedBlockMock),
   blockToBinary(signedBlockMock),
 ]
 
-class ChainArchiveMock {
+class ChainArchiveMock extends ChainArchive {
```

Comment:
No, it doesn't extend. This is a completely independent implementation of the same public API. Thus there should be an interface that both should implement.

Reply:
I tried this:
```
class ChainArchiveMock implements ChainArchive {
  private db: any;
  private readonly logger: Logger;
  private readonly path: string;

  private getBlockByNumber(blockNumber: number): Promise<Buffer> {
    return Promise.resolve(Buffer.from([]));
  }

  private async getLastBlockNumberFromDb(): Promise<number> {
    return Promise.resolve(666);
  }

  public async *getBlocks(_lastProcessedBlock: number): AsyncGenerator<ArchivedBlock, void> {
    for (let index = 0; index < blocksMock.length; index++) {
      const blockBuffer = blocksMock[index];

      yield new ArchivedBlock(blockBuffer, {
        number: index + 1,
        hash: '0xcf5fa8ef2fe76c0d6288535231d21989829933b986d32a3ba452173c5a2074f1',
      });
    }
  }
}
```
and I got TS compiler error:
```
Types have separate declarations of a private property 'db'
```
Do you know how can I fix that?

Reply:
You can't implement a class on a class. You need to create a separate interface and make both `ChainArchive` and `ChainArchiveMock` implement it.

Reply:
I still get same error.

```
interface IChainArchive {
  getBlocks(lastProcessedBlock: number): AsyncGenerator<ArchivedBlock, void>
}

class ChainArchiveMock implements IChainArchive {
  private db: any;
  private readonly logger: Logger;
  private readonly path: string;

  public constructor({ path, logger }: ChainArchiveConstructorParams) {
    this.path = path;
    this.logger = logger;
  }

  private getBlockByNumber(blockNumber: number): Promise<Buffer> {
    return Promise.resolve(Buffer.from([]));
  }

  private async getLastBlockNumberFromDb(): Promise<number> {
    return Promise.resolve(666);
  }

  public async *getBlocks(_lastProcessedBlock: number): AsyncGenerator<ArchivedBlock, void> {
    for (let index = 0; index < blocksMock.length; index++) {
      const blockBuffer = blocksMock[index];

      yield new ArchivedBlock(blockBuffer, {
        number: index + 1,
        hash: '0xcf5fa8ef2fe76c0d6288535231d21989829933b986d32a3ba452173c5a2074f1',
      });
    }
  }
}
```

Reply:
Which line? Did you update dependencies to rely on `IChainArchive` and not `ChainArchive`?

Reply:
yes, i got it now, thanks